### DISCUSSION TITLE
Shutter Operation using Power Metering

### DISCUSSION THREAD

**Discussion by kabongsteve** [arendst/Tasmota#15120] [Ideas]
The new Sonoff Dual R3 includes Power Metering as part of its functionality.
It is also Ideal for use as a Blind/Shutter or Awning controller with its Dual Relays, and dual switch connections.

Because it has power metering, this should mean it knows whether the blind/shutter is currently operating or not.

Thus instead of needing to set a timer for the duration, it could simply monitor the power usage, and when power usage stops (or drops significantly), then it knows the shutter has reached an end position and can turn off the relays and reset its position indicator.

This would also mean it could self configure any settings needed, and reset its position monitoring when it goes out of wack (like after power outage or controller reboot).

Thus the current ShutterEnableEndStopTime, ShutterOpenDuration, ShutterCloseDuration would not be needed

It may even be able to calculate the ShutterMotorDelay

**Comment by sfromis** [comment]
I'd expect the energy monitoring reaction time to be too slow to match what you can get by calibrating the durations.

**Reply by Mark-Muc** [reply]
i think it will be a good idea from [kabongsteve].. like Shelly uses this kind of "automatic calibration" in their org. firmware!!

**Comment by hacker-h** [comment]
The shelly original firmware automatically calibrates by one full movement up and down and then starts examining start and stop positions more closely. The total calibration took at least 3-4x up and down movement at each end for me.
In the end the precision of this movement after finished calibration is around +-3cm for me. So it is pretty close, but not perfect.

**Comment by stefanbode** [comment]
the TASMOTA can do the same on ESP32. See documentation. The only requirement is that the powermeter is fast enough, but this works for shelly. The callibration check for power change to detect end positions based on a 0.1 second accuracy
